server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/VacationMailetTest.java

```diff
@@ -109,10 +116,89 @@ public void activateVacationShouldSendNotification() throws Exception {
         when(zonedDateTimeProvider.get()).thenReturn(DATE_TIME_2017);
         when(automaticallySentMailDetector.isAutomaticallySent(mail)).thenReturn(false);
+        when(notificationRegistry.isRegistered(ACCOUNT_ID, recipientId))
+            .thenReturn(CompletableFuture.completedFuture(false));
 
         testee.service(mail);
 
         verify(mailetContext).sendMail(eq(originalRecipient), eq(ImmutableList.of(originalSender)), any());
+        verify(notificationRegistry).isRegistered(ACCOUNT_ID, recipientId);
+        verify(notificationRegistry).register(ACCOUNT_ID, recipientId, Optional.of(DATE_TIME_2018));
+        verifyNoMoreInteractions(mailetContext, notificationRegistry);
+    }
+
+    @Test
+    public void activateVacationShouldNotSendNotificationIfAlreadySent() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .fileName("spamMail.eml")
+            .recipient(originalRecipient)
+            .sender(originalSender)
+            .build();
+        when(vacationRepository.retrieveVacation(AccountId.fromString(USERNAME)))
+            .thenReturn(CompletableFuture.completedFuture(
+                Vacation.builder()
+                    .enabled(true)
+                    .fromDate(Optional.of(DATE_TIME_2016))
+                    .toDate(Optional.of(DATE_TIME_2018))
+                    .textBody("Explaining my vacation")
+                    .build()));
+        when(zonedDateTimeProvider.get()).thenReturn(DATE_TIME_2017);
+        when(notificationRegistry.isRegistered(ACCOUNT_ID, recipientId))
+            .thenReturn(CompletableFuture.completedFuture(true));
+
+        testee.service(mail);
+
+        verifyNoMoreInteractions(mailetContext);
+    }
+
+    @Test
+    public void activateVacationShouldSendNotificationIfErrorUpdatingNotificationRepository() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .fileName("spamMail.eml")
+            .recipient(originalRecipient)
+            .sender(originalSender)
+            .build();
+        when(vacationRepository.retrieveVacation(AccountId.fromString(USERNAME)))
+            .thenReturn(CompletableFuture.completedFuture(
+                Vacation.builder()
+                    .enabled(true)
+                    .fromDate(Optional.of(DATE_TIME_2016))
+                    .toDate(Optional.of(DATE_TIME_2018))
+                    .textBody("Explaining my vacation")
+                    .build()));
+        when(zonedDateTimeProvider.get()).thenReturn(DATE_TIME_2017);
+        RecipientId recipientId = RecipientId.fromMailAddress(originalSender);
+        when(notificationRegistry.isRegistered(ACCOUNT_ID, recipientId))
+            .thenReturn(CompletableFuture.completedFuture(true));
+        when(notificationRegistry.register(ACCOUNT_ID, recipientId, Optional.of(DATE_TIME_2018))).thenThrow(new RuntimeException());
+
+        testee.service(mail);
+
+        verifyNoMoreInteractions(mailetContext);
```

Comment:
Yes. Good catch.

The mail was not sent because of 

```
when(notificationRegistry.isRegistered(ACCOUNT_ID, recipientId))
            .thenReturn(CompletableFuture.completedFuture(true));
```

Should be false...